src/gptl/gptl.c

```diff
@@ -801,18 +1112,27 @@ int GPTLstart_handle (const char *name,  /* timer name */
   } else {
     ptr = getentry (hashtable[t], name, &indx);
   }
-    
-  /* 
-  ** Recursion => increment depth in recursion and return.  We need to return 
+
+  /*
+  ** Recursion => increment depth in recursion and return.  We need to return
   ** because we don't want to restart the timer.  We want the reported time for
   ** the timer to reflect the outermost layer of recursion.
   */
 
   if (ptr && ptr->onflg) {
-#ifdef GPTL_ASSERT_ON_RECURSION
-    assert(0);
-#endif
```

Comment:
Why did we lose this change?

Reply:
You are right, this lost change is in PR #351 and we need to reapply it. Will recheck other changes as well.